src/BayesFilters/include/BayesFilters/Gaussian.h

```diff
@@ -26,7 +26,7 @@ class bfl::Gaussian : public bfl::GaussianMixture
 
     Gaussian(const std::size_t dim_linear) noexcept;
 
-    Gaussian(const std::size_t dim_linear, const std::size_t dim_circular) noexcept;
+    Gaussian(const std::size_t dim_linear, const std::size_t dim_circular, const bool& use_quaternion = false) noexcept;
```

Comment:
Don't really like this ctor anymore. Is becoming uglier and uglier, but let's stick with it for now 😄.
For primitive data type, there is non need to take reference. `const bool` suffices.

Reply:
> Don't really like this ctor anymore. Is becoming uglier and uglier, but let's stick with it for now :smile:

Totally agree :laughing: I think that, once we are ready to drop Euler angles, we can remove the additional parameter and leave quaternions only (in the end, the literature is now full of algorithms for quaternions and lie groups). At some point, we will probably re-design the way the state vector is handled.